components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs

```diff
@@ -30,11 +30,56 @@ public class WriterGroupMessageTrigger : IMessageTrigger, IDisposable {
         public int NumberOfConnectionRetries => _subscriptions?.FirstOrDefault()?.
             Subscription?.NumberOfConnectionRetries ?? 0;
 
+        private IDictionary<DateTime, int> _valueChanges = new Dictionary<DateTime, int>(100);
         /// <inheritdoc/>
-        public int ValueChangesCount { get; private set; } = 0;
+        public int ValueChangesCountLastMinute
+        {
+            get
+            {
+                return _valueChanges.Values.Sum();
+            }
+            private set
+            {
+                var difference = Math.Abs(value);
+                var now = DateTime.UtcNow;
+                var oneMinuteTimespan = TimeSpan.FromMinutes(1);
+                _valueChanges.Add(now, difference);
+
+                var valuesToDelete = _valueChanges.Where(kvp => now - kvp.Key > oneMinuteTimespan).Select(kvp => kvp.Key);
+                foreach (var valueToDelete in valuesToDelete) {
+                    _valueChanges.Remove(valueToDelete);
+                }
+            }
+        }
+
+        private int _valueChangesCount = 0;
+        /// <inheritdoc/>
+        public int ValueChangesCount { get { return _valueChangesCount; } private set { var difference = (value - _valueChangesCount);  _valueChangesCount = value; ValueChangesCountLastMinute = difference; } }
```

Comment:
@koepalex, please break lines `57` and `82` as well into several lines if they are longer than `120` characters. It is kind of hard to tell on GitHub if they are or not.

Reply:
added line breaks